tests/foreman/cli/test_errata.py

```diff
@@ -80,113 +74,107 @@
+            'content-view-id': module_cv.id,
+            'lifecycle-environment-id': module_lce.id,
+            'activationkey-id': module_activation_key.id,
+        }
+    )
 
-        :BZ: 1405428, 1418026, 1457977
-        """
-        super().setUpClass()
-        cls.org = make_org()
-        cls.env = make_lifecycle_environment({'organization-id': cls.org['id']})
-        cls.content_view = make_content_view({'organization-id': cls.org['id']})
-        cls.activation_key = make_activation_key(
-            {'lifecycle-environment-id': cls.env['id'], 'organization-id': cls.org['id']}
-        )
-        # add subscription to Satellite Tools repo to activation key
-        setup_org_for_a_rh_repo(
-            {
-                'product': PRDS['rhel'],
-                'repository-set': REPOSET['rhst7'],
-                'repository': REPOS['rhst7']['name'],
-                'organization-id': cls.org['id'],
-                'content-view-id': cls.content_view['id'],
-                'lifecycle-environment-id': cls.env['id'],
-                'activationkey-id': cls.activation_key['id'],
-            }
-        )
-        # create custom repository and add subscription to activation key
-        setup_org_for_a_custom_repo(
-            {
-                'url': cls.CUSTOM_REPO_URL,
-                'organization-id': cls.org['id'],
-                'content-view-id': cls.content_view['id'],
-                'lifecycle-environment-id': cls.env['id'],
-                'activationkey-id': cls.activation_key['id'],
-            }
-        )
 
-    def setUp(self):
-        """Create and setup host collection, hosts and virtual machines."""
-        super().setUp()
-        self.virtual_machines = []
-        self.host_collection = make_host_collection({'organization-id': self.org['id']})
-        for _ in range(self.VIRTUAL_MACHINES_COUNT):
-            # create VM
-            virtual_machine = VirtualMachine(distro=DISTRO_RHEL7)
-            virtual_machine.create()
-            self.addCleanup(vm_cleanup, virtual_machine)
-            self.virtual_machines.append(virtual_machine)
-            virtual_machine.install_katello_ca()
-            # register content host
-            virtual_machine.register_contenthost(
-                self.org['name'], activation_key=self.activation_key['name']
-            )
-            # enable red hat satellite repository
-            virtual_machine.enable_repo(REPOS['rhst7']['id'])
-            # install katello-agent
-            virtual_machine.install_katello_agent()
-            host = Host.info({'name': virtual_machine.hostname})
-            HostCollection.add_host(
-                {
-                    'id': self.host_collection['id'],
-                    'organization-id': self.org['id'],
-                    'host-ids': host['id'],
-                }
-            )
-        ActivationKey.add_host_collection(
+@pytest.fixture(scope='module')
+def setUp(module_org, module_activation_key):
```

Comment:
Please use a descriptive name for the fixture function.  It doesn't appear to return anything, so maybe a name like `register_host_collection`.

Indentation wise - this is not indented, but defined under the class definition. Same as above, please make it class scoped and define within the class, or move it's definition out of the class.

Reply:
I have removed the classes and reformatted the file, added new fixtures to break up the larger ones